### PR TITLE
Mvp darjusch

### PR DESCRIPTION
**Pull request by Darjusch**
https://github.com/Darjusch/fitup/issues/1#issue-1158590687. -> getBet function #1
https://github.com/Darjusch/fitup/issues/8#issue-1158598154 -> addNgo #8

Refactored tests.

### CODE REVIEW

File: test/Fitup.js [hunk 1]
```diff
@@ -15,31 +15,31 @@ contract("Fitup", accounts => {
...
-        });
-        const before = Number(ethers.utils.formatEther(await web3.eth.getBalance(account_one)));
-        await fitup.payoutBet(true, account_one);
-        const after = Number(ethers.utils.formatEther(await web3.eth.getBalance(account_one)));
+        let {before, after} = await beforeAfterPayout(account_one, true) // bet won
         expect(before).to.be.lessThan(after);
     });
-})
\ No newline at end of file
+    async function beforeAfterPayout(from, result) {
```
Comment (5h3rr1ll) [inline test/Fitup.js:38]: strange function "beforeAfter..." 😀

**Review by 5h3rr1ll — APPROVED** [review]
Looks good to me.

File: test/Fitup.js (hunk 1 above)
Comment (Darjusch) [reply test/Fitup.js:38]: Haha true maybe beforeAndAfterPayout is a better naming?

File: contracts/Fitup.sol [hunk 2]
```diff
@@ -5,14 +5,21 @@ contract Fitup {
...
 
+    struct Ngo {
+        string name;
+        address organisation;
+    }
+
     struct Bet {
         uint256 amount;
         address organisation; 
     }
```
Comment (Nameer-Alkhayyat) [inline contracts/Fitup.sol:16]: Bet, the structure will need to include other variables that we agreed on.

File: contracts/Fitup.sol [hunk 3]
```diff
@@ -45,4 +53,9 @@ contract Fitup {
         }
         emit BetPayout(_issuer, _success, _organisation, _amount);
     }
+
+    function addNgo(string memory name, address organisation) external onlyOwner {
+        ngos.push(Ngo(name, organisation));
+        emit NgoAdded(name, organisation);
+    }
```
Comment (Nameer-Alkhayyat) [inline contracts/Fitup.sol:60]: I realized there is a problem in our design! How can we eliminate functions that might increase computation power? Also, if we add an ngo to NGO[] how we can delete it later?

**Review by Nameer-Alkhayyat — APPROVED** [review]
Yo, maybe let's discuss SC design once again!

File: contracts/Fitup.sol (hunk 2 above)
Comment (Darjusch) [reply contracts/Fitup.sol:16]: You are right, can you add this when you create the createBet function?

File: contracts/Fitup.sol (hunk 3 above)
Comment (Darjusch) [reply contracts/Fitup.sol:60]: I am not sure i understand the eliminate functions to save computational power part can you explain a bit more? 

For the second part, i also though we should have a updatengo and a deletengo function. 

Do you agree? Should we have a call?

**Pull request merged**